EIPS/eip-3540.md

```diff
@@ -93,88 +89,94 @@ The list of section headers is terminated with the *section headers terminator b
+| num_code_sections      | 2 bytes  | 0x0001-0xFFFF | 16-bit unsigned big-endian integer denoting the number of the code sections             |
+| code_size              | 2 bytes  | 0x0001-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the code section content      |
+| kind_container         | 1 byte   | 0x03          | kind marker for container size section                                                  |
+| num_container_sections | 2 bytes  | 0x0001-0x00FF | 16-bit unsigned big-endian integer denoting the number of the container sections        |
+| container_size         | 2 bytes  | 0x0001-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the container section content |
+| kind_data              | 1 byte   | 0x04          | kind marker for data size section                                                       |
+| data_size              | 2 bytes  | 0x0000-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the data section content (*)  |
+| terminator             | 1 byte   | 0x00          | marks the end of the header                                                             |
+
+(*) For not yet deployed containers this can be more than the actual content, see Data Section Lifecycle section in the contract creation EIP-???? (TBD + link).
+
+#### Body
+
+| name              | length   | value         | description                                                                           |
+|-------------------|----------|---------------|---------------------------------------------------------------------------------------|
+| types_section     | variable | n/a           | stores code section metadata                                                          |
+| inputs            | 1 byte   | 0x00-0x7F     | number of stack elements the code section consumes                                    |
+| outputs           | 1 byte   | 0x00-0x80     | number of stack elements the code section returns or 0x80 for non-returning functions |
+| max_stack_height  | 2 bytes  | 0x0000-0x03FF | maximum number of elements ever placed onto the stack by the code section             |
+| code_section      | variable | n/a           | arbitrary sequence of bytes                                                           |
+| container_section | variable | n/a           | arbitrary sequence of bytes                                                           |
+| data_section      | variable | n/a           | arbitrary sequence of bytes                                                           |
 
 See [EIP-4750](./eip-4750.md) for more information on the type section content.
+See [EIP-6206](./eip-6206.md) for more information on non-returning functions.
 
 #### EOF version 1 validation rules
 
-1. In addition to general validation rules above, EOF version 1 bytecode conforms to the rules specified below:
```

Comment:
This part was mandating exact order of the sections, we are kind of missing it now, but maybe the formula in `Container` section is enough.

Reply:
yea, I assumed it's enough and much cleaner than this